units/src/result.rs

```diff
@@ -87,6 +87,36 @@ pub enum NumOpResult<T> {
     Error(NumOpError),
 }
 
+impl<T> NumOpResult<T> {
+
+    /// Maps a `NumOpResult<T>` to `NumOpResult<U>` by applying a function to a
+    /// contained [`NumOpResult::Valid`] value, leaving a [`NumOpResult::Error`] value untouched.
```

Comment:
```suggestion
    /// Maps a `NumOpResult<T>` to `NumOpResult<U>`.
    ///
    /// Applies a function to the contained [`NumOpResult::Valid`] value, leaving the
    /// [`NumOpResult::Error`] value untouched i.e., returning the first error.
```

I acked but perhaps this is better?

Reply:
sure, the added clarification could be helpful

Reply:
Oooph, bad reviewing Tobin. Your original text is almost identical to `std::result::Result::map` - I owe you a drink for that one.

Reply:
Oh, I know what "first error" means but didn't realize it doesn't make that much sense in this context.

Reply:
same ^

I've changed it back though